### ISSUE TITLE
`obj.await` should not throw error.

### ISSUE THREAD

**Issue by mysticatea** [eslint/espree#258]
From https://github.com/eslint/eslint/issues/5564
On v3.1.1.

In `obj.await` case, `await` should not throw a syntax error because this `await` is `IdentifierName` (a reserved word is allowed).
Acorn has succeeded to parse it.
I guess this error caused by [this line](https://github.com/eslint/espree/blob/master/espree.js#L115).

```js
"use strict";

const espree = require("espree");
const ast = espree.parse(
    `obj.await`,
    {
        ecmaVersion: 6,
        sourceType: "module"
    }
);

console.log(JSON.stringify(ast, null, 4));
```

```
SyntaxError: Unexpected token await
    at Parser.instance.raise.instance.raiseRecoverable (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\espree\espree.js:394:19)
    at Parser.instance.unexpected (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\espree\espree.js:427:14)
    at Parser.esprimaFinishNode (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\espree\espree.js:142:14)
    at Parser.finishNode (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\espree\espree.js:232:34)
    at Parser.pp.parseIdent (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:678:15)
    at Parser.pp.parseSubscripts (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:236:28)
    at Parser.pp.parseExprSubscripts (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:228:15)
    at Parser.pp.parseMaybeUnary (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:204:17)
    at Parser.pp.parseExprOps (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:151:19)
    at Parser.pp.parseMaybeConditional (C:\Users\t-nagashima.AD\Documents\GitHub\eslint\node_modules\acorn\dist\acorn.js:133:19)
```

**Comment by nzakas** [comment]
I think you're right. I did this initially because Acorn wasn't properly handling `await` as a reserved word in modules. Maybe it's OK now.

**Comment by xjamundx** [comment]
I *may* be able to look at this this week assuming it's removing that check and just making sure the tests don't break

**Comment by nzakas** [comment]
Should be

**Comment by nzakas** [comment]
I'm in here looking at other changes, so I'm going to take a look at this while I have the hood up.
